indico/modules/events/abstracts/controllers/abstract.py

```diff
@@ -68,10 +69,39 @@ def _checkParams(self, params):
         RHAbstractBase._checkParams(self, params)
         self.abstract_file = AbstractFile.get_one(request.view_args['file_id'])
 
+
+class RHAbstractDownloadAttachment(RHAbstractAttachmentBase):
+    """Download an attachment file belonging to an abstract."""
+
     def _process(self):
         return self.abstract_file.send()
 
 
+class RHAbstractDeleteAttachment(RHAbstractAttachmentBase):
+    """Delete an attachment file belonging to an abstract."""
+
+    def _check_abstract_protection(self):
+        return self.abstract.can_edit(session.user)
+
+    def _process(self):
+        db.session.delete(self.abstract_file)
```

Comment:
Don't we want to add logging here? In that case, it's better to move this to `operations.py`.